src/test/java/net/openhft/chronicle/threads/MediumEventLoopTest.java

```diff
@@ -103,10 +104,67 @@ public void loopStarted() {
                 public boolean action() {
                     return false;
                 }
+
+                @Override
+                public void loopFinished() {
+                    throw new IllegalStateException("Something went wrong in loopFinished!!!");
+                }
+            }
+
+            try {
+                // Add handler before loop has started. loopStarted not called yet.
+                eventLoop.addHandler(new ThrowingHandler());
+
+                // Start the loop. loopStarted called.
+                eventLoop.start();
+
+                while (!eventLoop.isStarted()) {
+                    Thread.sleep(100);
+                }
+
+                expectException("Something went wrong in loopStarted!!!");
+                expectException("Something went wrong in loopFinished!!!");
+            } catch (Throwable t) {
+                fail("Unexpected exception", t);
+            }
+        }
+    }
+
+    @Test
+    void throwableExceptionsAreLoggedWhenThrownInLoopStarted() throws InterruptedException {
+        try (MediumEventLoop eventLoop = new MediumEventLoop(null, "name", Pauser.balanced(), true, null)) {
+            class ThrowingHandler implements EventHandler {
+
+                @Override
+                public void loopStarted() {
+                    throw new IllegalStateException("Something went wrong in loopStarted!!!");
+                }
+
+                @Override
+                public boolean action() {
+                    return false;
+                }
+
+                @Override
+                public void loopFinished() {
+                    System.out.println("finished");
+                }
             }
-            eventLoop.addHandler(new ThrowingHandler());
             eventLoop.start();
+            while(!eventLoop.isStarted()) {
+                Thread.sleep(100);
+            }
+
+            // Add handler after loop has started
             expectException("Something went wrong in loopStarted!!!");
+            eventLoop.addHandler(new ThrowingHandler());
+
+            // Not ideal, but we need to wait for the exception to be thrown
+            Thread.sleep(1000);
```

Comment:
Hi @david-ry4n in the test framework there is a `Waiters` abstraction that could be used here, it can wait until a given condition is met

Reply:
Thanks. Fixed.